src/schema/mutation/editResource.mutation.ts

```diff
@@ -666,36 +690,78 @@ export default {
 
       // Updating field permissions
       if (args.fieldsPermissions) {
-        const permissions: FieldPermissionChange = args.fieldsPermissions;
-        for (const permission in permissions) {
-          const obj: SimpleFieldPermissionChange = permissions[permission];
-          // Add permission on target field
-          if (obj.add) {
-            checkFieldPermission(
-              context,
-              get(resource, 'permissions'),
-              allResourceFields,
-              obj.add.field,
-              obj.add.role,
-              permission
-            );
-            addFieldPermission(
-              update,
-              allResourceFields,
-              obj.add.field,
-              obj.add.role,
-              permission
-            );
+        if (!args.multipleFields) {
+          const permissions: FieldPermissionChange = args.fieldsPermissions;
+          for (const permission in permissions) {
+            const obj: SimpleFieldPermissionChange = permissions[permission];
+            // Add permission on target field
+            if (obj.add) {
+              checkFieldPermission(
+                context,
+                get(resource, 'permissions'),
+                allResourceFields,
+                obj.add.field,
+                obj.add.role,
+                permission
+              );
+              addFieldPermission(
+                update,
+                allResourceFields,
+                obj.add.field,
+                obj.add.role,
+                permission
+              );
+            }
+            // Remove permission on target field
+            if (obj.remove) {
+              removeFieldPermission(
+                update,
+                allResourceFields,
+                obj.remove.field,
+                obj.remove.role,
+                permission
+              );
+            }
           }
-          // Remove permission on target field
-          if (obj.remove) {
-            removeFieldPermission(
-              update,
-              allResourceFields,
-              obj.remove.field,
-              obj.remove.role,
-              permission
-            );
+        } else {
+          const permissions: FieldsPermissionsChange = args.fieldsPermissions;
+          for (const permission in permissions) {
+            const obj: MultipleFieldsPermissionsChange =
+              permissions[permission];
+
+            // Add permission on target fields
+            if (obj.add) {
+              obj.add.fields.forEach((field) => {
+                checkFieldPermission(
+                  context,
+                  get(resource, 'permissions'),
+                  allResourceFields,
+                  field,
+                  obj.add.role,
+                  permission,
+                  !!permissions.canSee.add
```

Comment:
This is fine, a little suggestion that could be more readable
```suggestion
                  ('canSee' in permissions && 'add' in permissions.canSee)
```